test/e2e/cronfederatedhpa_test.go

```diff
@@ -0,0 +1,154 @@
+	"k8s.io/utils/pointer"
+
+	autoscalingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/autoscaling/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+/*
+CronFederatedHPA focus on scaling FederatedHPA or other resource with scale subresource (e.g. Deployment, StatefulSet).
+Test Case Overview:
+
+	case 1:
+		Scale FederatedHPA.
+	case 2:
+		Scale deployment.
+	case 3:
+		Test suspend rule with CronFederatedHPA
+*/
+var _ = ginkgo.Describe("[CronFederatedHPA] CronFederatedHPA testing", func() {
+	var cronFHPAName, fhpaName, policyName, deploymentName string
+	var cronFHPA *autoscalingv1alpha1.CronFederatedHPA
+	var fhpa *autoscalingv1alpha1.FederatedHPA
+	var deployment *appsv1.Deployment
+	var policy *policyv1alpha1.PropagationPolicy
+
+	ginkgo.BeforeEach(func() {
+		cronFHPAName = cronFedratedHPANamePrefix + rand.String(RandomStrLength)
+		fhpaName = federatedHPANamePrefix + rand.String(RandomStrLength)
+		policyName = deploymentNamePrefix + rand.String(RandomStrLength)
+		deploymentName = policyName
+
+		deployment = helper.NewDeployment(testNamespace, deploymentName)
+		policy = helper.NewPropagationPolicy(testNamespace, policyName, []policyv1alpha1.ResourceSelector{
+			{
+				APIVersion: deployment.APIVersion,
+				Kind:       deployment.Kind,
+				Name:       deploymentName,
+			},
+		}, policyv1alpha1.Placement{
+			ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+				ClusterNames: framework.ClusterNames(),
+			},
+			ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+				ReplicaSchedulingType:     policyv1alpha1.ReplicaSchedulingTypeDivided,
+				ReplicaDivisionPreference: policyv1alpha1.ReplicaDivisionPreferenceAggregated,
+			},
+		})
+	})
+
+	ginkgo.JustBeforeEach(func() {
+		framework.CreatePropagationPolicy(karmadaClient, policy)
+		framework.CreateDeployment(kubeClient, deployment)
+		ginkgo.DeferCleanup(func() {
+			framework.RemoveDeployment(kubeClient, deployment.Namespace, deployment.Name)
+			framework.RemovePropagationPolicy(karmadaClient, policy.Namespace, policy.Name)
+		})
+	})
+
+	// case 1: Scale FederatedHPA.
+	ginkgo.Context("Scale FederatedHPA", func() {
+		ginkgo.BeforeEach(func() {
+			rule := helper.NewCronFederatedHPARule("scale-up", "*/1 * * * *", false, nil, pointer.Int32(2), pointer.Int32(100))
+			fhpa = helper.NewFederatedHPA(testNamespace, fhpaName, deploymentName)
+			cronFHPA = helper.NewCronFederatedHPAWithScalingFHPA(testNamespace, cronFHPAName, fhpaName, rule)
+
+			framework.CreateFederatedHPA(karmadaClient, fhpa)
+		})
+
+		ginkgo.AfterEach(func() {
+			framework.RemoveFederatedHPA(karmadaClient, testNamespace, fhpaName)
+			framework.RemoveCronFederatedHPA(karmadaClient, testNamespace, cronFHPAName)
+		})
+
+		ginkgo.It("Scale FederatedHPA testing", func() {
+			// Wait FederatedHPA to scale deployment's replicas to 1
```

Comment:
How do you ensure that  the replicas of the deployment can scale-down to 1 when I see that you have set MinReplicas to 1,  MaxReplicas to 10 and AverageUtilization to 80 for FederatedHPA? 


Is there such a situation where `nginx` may consume some CPU, causing the replicas to only scale-down to 2?

Reply:
We do not request nginx pods, I think scaling down the replicas of nginx is certain.

Reply:
How about setting  MinReplicas to 1 and MaxReplicas to 1 ?

Reply:
It's better idea.